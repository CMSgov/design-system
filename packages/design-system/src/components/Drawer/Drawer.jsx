import FocusTrap from 'focus-trap-react';
import Button from '../Button/Button';
import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';
import uniqueId from 'lodash/uniqueId';

export class Drawer extends React.PureComponent {
  constructor(props) {
    super(props);
    this.headingRef = null;
    this.id = this.props.headingId || uniqueId('drawer_');
    this.handleEscapeKey = this.handleEscapeKey.bind(this);
  }

  componentDidMount() {
    if (this.props.hasFocusTrap) document.addEventListener('keydown', this.handleEscapeKey);
    if (this.headingRef) this.headingRef.focus();
  }

  componentWillUnmount() {
    document.removeEventListener('keydown', this.handleEscapeKey);
  }

  handleEscapeKey(evt) {
    switch (evt.code) {
      case 'Escape':
        this.props.onCloseClick();
        break;
      default:
        break;
    }
  }

  render() {
    const {
      ariaLabel,
      children,
      className,
      closeButtonText,
      footerBody,
      footerTitle,
      hasFocusTrap,
      heading,
      isHeaderSticky,
      isFooterSticky,
      onCloseClick,
    } = this.props;
    const Heading = `h${this.props.headingLevel}` || `h3`;

    /* eslint-disable jsx-a11y/no-noninteractive-tabindex, react/no-danger */
    const drawerMarkup = () => (
      <div aria-labelledby={this.id} className={classNames(className, 'ds-c-drawer')} role="dialog">
        <div className="ds-c-drawer__window">
          <div className="ds-c-drawer__header">
            <Heading
              tabIndex="0"
              id={this.id}
              className="ds-c-drawer__header-heading"
              ref={(el) => (this.headingRef = el)}
            >
              {heading}
            </Heading>
            <Button
              aria-label={ariaLabel}
              className="ds-c-drawer__close-button"
              size="small"
              onClick={onCloseClick}
            >
              {closeButtonText}
            </Button>
          </div>
          <div
            className={classNames('ds-c-drawer__body', {
              'ds-c-drawer--is-sticky': isHeaderSticky || isFooterSticky,
            })}
          >
            {children}
          </div>
          <div className="ds-c-drawer__footer">
            <h4 className="ds-c-drawer__footer-title">{footerTitle}</h4>
            <div className="ds-c-drawer__footer-body">{footerBody}</div>
          </div>
        </div>
      </div>
    );

    return (
      <>
        {hasFocusTrap ? (
          <FocusTrap focusTrapOptions={{ clickOutsideDeactivates: true }}>
            {drawerMarkup()}
          </FocusTrap>
        ) : (
          drawerMarkup()
        )}
      </>
    );
  }
}

Drawer.defaultProps = {
  ariaLabel: 'Close help drawer',
  closeButtonText: 'Close',
  headingLevel: '3',
};

// TODO: closeButtonText, heading should be a string, but it is being used as a node in MCT,
// until we provide a better solution for customization, we type it as a node.
Drawer.propTypes = {
  /**
   * Gives more context to screen readers on the Drawer close button.
   */
  ariaLabel: PropTypes.string,
  closeButtonText: PropTypes.node,
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  footerBody: PropTypes.node,
  footerTitle: PropTypes.string,
  /**
   * Enables focus trap functionality within Drawer.
   */
  hasFocusTrap: PropTypes.bool,
  /**
   * Text for the Drawer heading. Required because the `heading` will be focused on mount.
   */
  heading: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
  /**
   * A unique `id` to be used on heading element to label multiple instances of Drawer.
   */
  headingId: PropTypes.string,
  /**
   * Heading type to override default `<h3>`
   */
  headingLevel: PropTypes.oneOf(['1', '2', '3', '4', '5']),
  /**
   * Enables "sticky" position of Drawer header element.
   */
  isHeaderSticky: PropTypes.bool,
  /**
   * Enables "sticky" position of Drawer footer element.
   */
  isFooterSticky: PropTypes.bool,
  onCloseClick: PropTypes.func.isRequired,
};

export default Drawer;
