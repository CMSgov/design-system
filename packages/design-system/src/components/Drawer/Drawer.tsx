import Button from '../Button/Button';
import NativeDialog from '../NativeDialog/NativeDialog';
import { useRef } from 'react';
import classNames from 'classnames';
import uniqueId from 'lodash/uniqueId';
import { t } from '../i18n';

// TODO: closeButtonText, heading should be a string, but it is being used as a node in MCT,
// until we provide a better solution for customization, we type it as a node.
export interface DrawerProps {
  /**
   * Gives more context to screen readers on the Drawer close button.
   */
  ariaLabel?: string;
  closeButtonText?: React.ReactNode;
  children: React.ReactNode;
  className?: string;
  footerBody?: React.ReactNode;
  footerTitle?: string;
  /**
   * Enables focus trap functionality within Drawer.
   */
  hasFocusTrap?: boolean;
  /**
   * Text for the Drawer heading. Required because the `heading` will be focused on mount.
   */
  heading: string | React.ReactNode;
  /**
   * A unique `id` to be used on heading element to label multiple instances of Drawer.
   */
  headingId?: string;
  /**
   * Heading type to override default `<h3>`
   */
  headingLevel?: '1' | '2' | '3' | '4' | '5';
  /**
   * Ref to heading element
   */
  headingRef?: React.MutableRefObject<any>;
  /**
   * Enables "sticky" position of Drawer header element.
   */
  isHeaderSticky?: boolean;
  /**
   * Enables "sticky" position of Drawer footer element.
   */
  isFooterSticky?: boolean;
  onCloseClick: (event: React.MouseEvent | React.KeyboardEvent) => void;
}

export const Drawer = (props: DrawerProps) => {
  const headingRef = useRef(null);
  const id = useRef(props.headingId || uniqueId('drawer_'));

  const Heading = `h${props.headingLevel}` as const;

  return (
    <NativeDialog
      className={classNames(props.className, 'ds-c-drawer')}
      exit={props.onCloseClick}
      showModal={props.hasFocusTrap}
    >
      <div className="ds-c-drawer__window" tabIndex={-1} aria-labelledby={id.current}>
        <div className="ds-c-drawer__header">
          <Heading
            id={id.current}
            className="ds-c-drawer__header-heading"
            ref={(el) => {
              headingRef.current = el;
              if (props.headingRef) {
                props.headingRef.current = el;
              }
            }}
          >
            {props.heading}
          </Heading>
          <Button
            aria-label={props.ariaLabel ?? t('drawer.ariaLabel')}
            className="ds-c-drawer__close-button"
            size="small"
            onClick={props.onCloseClick}
          >
            {props.closeButtonText ?? t('drawer.closeButtonText')}
          </Button>
        </div>
        <div
          className={classNames('ds-c-drawer__body', {
            'ds-c-drawer--is-sticky': props.isHeaderSticky || props.isFooterSticky,
          })}
          tabIndex={0}
        >
          {props.children}
        </div>
        <div className="ds-c-drawer__footer">
          <h4 className="ds-c-drawer__footer-title">{props.footerTitle}</h4>
          <div className="ds-c-drawer__footer-body">{props.footerBody}</div>
        </div>
      </div>
    </NativeDialog>
  );
};

Drawer.defaultProps = {
  headingLevel: '3',
  hasFocusTrap: false,
};

export default Drawer;
