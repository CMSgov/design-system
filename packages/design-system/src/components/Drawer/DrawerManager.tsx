import React from 'react';
import { createContext, useState, useContext } from 'react';
import useId from '../utilities/useId';

export const DrawerContext = createContext(null);

export const DrawerManager = (props: any) => {
  const [currentID, setCurrentID] = useState(null);

  return <DrawerContext.Provider value={{ currentID, setCurrentID }} {...props} />;
};

export const useDrawerManager = () => {
  const { currentID, setCurrentID } = useContext(DrawerContext);
  const id = useId();

  const isOpen = currentID === id;
  const toggleClick = () => setCurrentID(isOpen ? null : id);
  const closeClick = () => setCurrentID(null);

  return { toggleClick, closeClick, isOpen };
};

export default DrawerManager;
