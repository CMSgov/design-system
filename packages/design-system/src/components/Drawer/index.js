export { default as Drawer } from './Drawer';
export { default as DrawerToggle } from './DrawerToggle';
