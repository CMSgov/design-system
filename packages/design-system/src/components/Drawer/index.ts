export * from './Drawer';
export * from './DrawerToggle';
// Can't release it right now with a focus-management bug
// export * from './DrawerManager';
