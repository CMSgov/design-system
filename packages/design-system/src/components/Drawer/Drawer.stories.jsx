/* eslint-disable react/display-name */
/* eslint-disable react/no-multi-comp */
import React from 'react';
import { Title, Subtitle, Description, ArgsTable, PRIMARY_STORY } from '@storybook/addon-docs';
import { useArgs } from '@storybook/client-api';

import Drawer from './Drawer';
import DrawerToggle from './DrawerToggle';

export default {
  title: 'Components/Drawer',
  component: Drawer,
  argTypes: {
    closeButtonText: {
      control: { type: 'text' },
    },
  },
  args: {
    footerTitle: 'Footer Title',
    footerBody: <p className="ds-text ds-u-margin--0">Footer content</p>,
    heading: 'Drawer Heading',
  },
  subcomponents: { DrawerToggle },
  // The Drawer was overlapping the docs page, so customizing the docs page to remove the examples
  parameters: {
    docs: {
      page: () => (
        <>
          <Title />
          <Subtitle />
          <Description />
          <ArgsTable story={PRIMARY_STORY} />
        </>
      ),
    },
  },
};

const drawerContent = (
  <>
    <strong>An Explanation</strong>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut
      labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco
      laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in
      voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat
      non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor
      sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore
      magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
      aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit
      esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident,
      sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet,
      consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna
      aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex
      ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
      dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa
      qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur
      adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad
      minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
      consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu
      fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
      officia deserunt mollit anim id est laborum.
    </p>
  </>
);

const Template = ({ data, ...args }) => <Drawer {...args}>{drawerContent}</Drawer>;

export const DrawerDefault = Template.bind({});
export const DrawerWithStickyPositioning = Template.bind({});
DrawerWithStickyPositioning.args = {
  isFooterSticky: true,
  isHeaderSticky: true,
};

export const DrawerToggleWithDrawer = () => {
  const [{ isDrawerVisible }, setIsDrawerVisible] = useArgs();

  return (
    <>
      {isDrawerVisible && (
        <Drawer
          onCloseClick={() => setIsDrawerVisible({ isDrawerVisible: false })}
          footerTitle="Footer Title"
          footerBody={<p className="ds-text ds-u-margin--0">Footer content</p>}
          heading="Drawer Heading"
        >
          {drawerContent}
        </Drawer>
      )}
      <DrawerToggle
        showDrawer={() => setIsDrawerVisible({ isDrawerVisible: true })}
        drawerOpen={isDrawerVisible || false}
      >
        Drawer Toggle
      </DrawerToggle>
    </>
  );
};
