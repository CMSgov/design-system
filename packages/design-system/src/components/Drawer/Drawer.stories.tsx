import React from 'react';
import { Title, Subtitle, Description, ArgsTable, PRIMARY_STORY } from '@storybook/addon-docs';
import { action } from '@storybook/addon-actions';
import Drawer from './Drawer';
import { Button } from '../Button';
import type { Meta, StoryObj } from '@storybook/react';
import { useArgs } from '@storybook/preview-api';

const meta: Meta<typeof Drawer> = {
  title: 'Components/Drawer',
  component: Drawer as any,
  args: {
    footerTitle: 'Footer Title',
    footerBody: <p className="ds-text ds-u-margin--0">Footer content</p>,
    heading: 'Drawer Heading',
  },
  // The Drawer was overlapping the docs page, so customizing the docs page to remove the examples
  parameters: {
    docs: {
      page: () => (
        <>
          <Title />
          <Subtitle />
          <Description />
          <ArgsTable story={PRIMARY_STORY} />
        </>
      ),
    },
  },
};
export default meta;

type Story = StoryObj<typeof Drawer>;

const drawerContent = (
  <>
    <strong>An Explanation</strong>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut
      labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco
      laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in
      voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat
      non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor
      sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore
      magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
      aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit
      esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident,
      sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet,
      consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna
      aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex
      ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
      dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa
      qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur
      adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad
      minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
      consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu
      fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
      officia deserunt mollit anim id est laborum.
    </p>
  </>
);

export const DrawerDefault: Story = {
  render: function Component(args) {
    return <Drawer {...args}>{drawerContent}</Drawer>;
  },
};
export const DrawerWithStickyPositioning: Story = {
  ...DrawerDefault,
  args: {
    isFooterSticky: true,
    isHeaderSticky: true,
  },
};

export const DrawerToggleWithDrawer: Story = {
  render: function Component() {
    const [{ isDrawerVisible, ...args }, updateArgs] = useArgs();
    const showDrawer = () => updateArgs({ isDrawerVisible: true });
    const hideDrawer = (...params) => {
      action('onCloseClick')(...params);
      updateArgs({ isDrawerVisible: false });
    };

    return (
      <>
        {isDrawerVisible && (
          <Drawer
            {...args}
            onCloseClick={hideDrawer}
            footerTitle="Footer Title"
            footerBody={<p className="ds-text ds-u-margin--0">Footer content</p>}
            heading="Drawer Heading"
          >
            {drawerContent}
          </Drawer>
        )}
        <Button className="ds-c-drawer__toggle" variation="ghost" onClick={showDrawer}>
          Drawer Toggle
        </Button>
      </>
    );
  },
};
