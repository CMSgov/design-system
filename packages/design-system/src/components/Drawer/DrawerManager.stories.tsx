import React from 'react';
import { Title, Subtitle, Description } from '@storybook/addon-docs';

import Drawer from './Drawer';
import { DrawerManager, useDrawerManager } from './DrawerManager';
import { Button } from '../Button';
import type { Meta, StoryObj } from '@storybook/react';

const meta: Meta<typeof DrawerManager> = {
  title: 'Components/DrawerManager',
  component: DrawerManager,
  parameters: {
    docs: {
      page: () => (
        <>
          <Title />
          <Subtitle />
          <Description />
        </>
      ),
    },
  },
};
export default meta;

type Story = StoryObj<typeof DrawerManager>;

const drawerContent1 = {
  heading: 'Drawer Number One',
  children: (
    <>
      <strong>Yar!</strong>
      <p>
        Prow scuttle parrel provost Sail ho shrouds spirits boom mizzenmast yardarm. Pinnace
        holystone mizzenmast quarter crow&apos;s nest nipperkin grog yardarm hempen halter furl.
        Swab barque interloper chantey doubloon starboard grog black jack gangway rutters.
      </p>

      <p>
        Deadlights jack lad schooner scallywag dance the hempen jig carouser broadside cable strike
        colors. Bring a spring upon her cable holystone blow the man down spanker Shiver me timbers
        to go on account lookout wherry doubloon chase. Belay yo-ho-ho keelhaul squiffy black spot
        yardarm spyglass sheet transom heave to.
      </p>
    </>
  ),
};

const drawerContent2 = {
  heading: 'Drawer Number Two',
  children: (
    <>
      <strong>The Cosmos Awaits</strong>
      <p>
        Radio telescope quasar bits of moving fluff Sea of Tranquility Drake Equation rogue? Network
        of wormholes descended from astronomers cosmic ocean a mote of dust suspended in a sunbeam
        from which we spring hearts of the stars? Preserve and cherish that pale blue dot stirred by
        starlight descended from astronomers of brilliant syntheses two ghostly white figures in
        coveralls and helmets are softly dancing hundreds of thousands? Vastness is bearable only
        through love invent the universe a still more glorious dawn awaits dream of the mind&apos;s
        eye not a sunrise but a galaxyrise courage of our questions and billions upon billions upon
        billions upon billions upon billions upon billions upon billions.
      </p>
    </>
  ),
};

const drawerContent3 = {
  heading: 'Drawer Number Three',
  children: (
    <>
      <strong>Space, the final frontier</strong>
      <p>
        In the last 24 hours we have seen facilities now being created for the greatest and most
        complex exploration in man&apos;s history. We have felt the ground shake and the air
        shattered by the testing of a Saturn C-1 booster rocket, many times as powerful as the Atlas
        which launched John Glenn, generating power equivalent to 10,000 automobiles with their
        accelerators on the floor. We have seen the site where the F-1 rocket engines, each one as
        powerful as all eight engines of the Saturn combined, will be clustered together to make the
        advanced Saturn missile, assembled in a new building to be built at Cape Canaveral as tall
        as a 48 story structure, as wide as a city block, and as long as two lengths of this field.
      </p>
    </>
  ),
};

const SingleDrawerWithToggle = (...args) => {
  const { heading, children } = args[0];
  const { toggleClick, closeClick, isOpen } = useDrawerManager();

  return (
    <>
      {isOpen && (
        <Drawer
          {...args}
          onCloseClick={closeClick}
          footerTitle="Footer Title"
          footerBody={<p className="ds-text ds-u-margin--0">Footer content</p>}
          heading={heading}
        >
          {children}
        </Drawer>
      )}
      <Button
        className="ds-c-drawer__toggle ds-u-margin-bottom--2"
        variation="ghost"
        onClick={toggleClick}
      >
        {heading}
      </Button>
    </>
  );
};

export const DrawerManagerDefault: Story = {
  render: function Component() {
    return (
      <DrawerManager>
        <SingleDrawerWithToggle {...drawerContent1} />
        <SingleDrawerWithToggle {...drawerContent2} />
        <SingleDrawerWithToggle {...drawerContent3} />
      </DrawerManager>
    );
  },
};
