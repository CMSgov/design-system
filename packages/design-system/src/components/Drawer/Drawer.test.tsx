import Drawer from './Drawer';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

const defaultProps = {
  footerBody: (
    <div>
      <p>Some footer content</p>
    </div>
  ),
  footerTitle: 'Footer title',
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  onCloseClick: () => {},
  heading: 'Drawer title',
};

function renderDrawer(props) {
  props = Object.assign({}, defaultProps, props);
  const wrapper = shallow(
    <Drawer {...props}>
      <p>content</p>
    </Drawer>
  );
  return { props, wrapper };
}

describe('Drawer', () => {
  it('calls props.onCloseClick on close button click', () => {
    const onCloseClick = jest.fn();
    const { wrapper } = renderDrawer({ onCloseClick });
    const closeBtn = wrapper.find('Button');
    closeBtn.simulate('click');
    expect(onCloseClick).toHaveBeenCalled();
  });

  it('renders a snapshot', () => {
    const tree = renderer
      .create(
        <Drawer {...defaultProps}>
          <p>content</p>
        </Drawer>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
