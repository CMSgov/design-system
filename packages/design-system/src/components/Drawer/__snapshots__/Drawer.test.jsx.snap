// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer renders a snapshot 1`] = `
<div
  aria-labelledby="drawer_2"
  className="ds-c-drawer"
  role="dialog"
>
  <div
    className="ds-c-drawer__window"
  >
    <div
      className="ds-c-drawer__header"
    >
      <h3
        className="ds-c-drawer__header-heading"
        id="drawer_2"
        tabIndex="0"
      >
        Drawer title
      </h3>
      <button
        aria-label="Close help drawer"
        className="ds-c-button ds-c-button--small ds-c-drawer__close-button"
        onClick={[Function]}
        type="button"
      >
        Close
      </button>
    </div>
    <div
      className="ds-c-drawer__body"
    >
      <p>
        content
      </p>
    </div>
    <div
      className="ds-c-drawer__footer"
    >
      <h4
        className="ds-c-drawer__footer-title"
      >
        Footer title
      </h4>
      <div
        className="ds-c-drawer__footer-body"
      >
        <div>
          <p>
            Some footer content
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
